diff -r -u a/src/connman-wait-online.service.in b/src/connman-wait-online.service.in
--- a/src/connman-wait-online.service.in	2015-12-08 15:02:25.000000000 +0200
+++ b/src/connman-wait-online.service.in	2015-12-08 15:18:19.061034581 +0200
@@ -1,7 +1,7 @@
 [Unit]
 Description=Wait for network to be configured by ConnMan
-Requisite=connman.service
-After=connman.service
+Requisite=connman-upstream.service
+After=connman-upstream.service
 Before=network-online.target
 DefaultDependencies=no
 Conflicts=shutdown.target
